core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -0,0 +1,633 @@
+                  typename ReducerType::value_type& /*value*/) const noexcept {
+    // FIXME_SYCL
+    Kokkos::abort("Not implemented!");
+  }
+
+  //--------------------------------------------------------------------------
+  /** \brief  Intra-team exclusive prefix sum with team_rank() ordering
+   *          with intra-team non-deterministic ordering accumulation.
+   *
+   *  The global inter-team accumulation value will, at the end of the
+   *  league's parallel execution, be the scan's total.
+   *  Parallel execution ordering of the league's teams is non-deterministic.
+   *  As such the base value for each team's scan operation is similarly
+   *  non-deterministic.
+   */
+  template <typename Type>
+  KOKKOS_INLINE_FUNCTION Type team_scan(const Type& value,
+                                        Type* const /*global_accum*/) const {
+    // FIXME_SYCL
+    Kokkos::abort("Not implemented!");
+    return value;
+  }
+
+  /** \brief  Intra-team exclusive prefix sum with team_rank() ordering.
+   *
+   *  The highest rank thread can compute the reduction total as
+   *    reduction_total = dev.team_scan( value ) + value ;
+   */
+  template <typename Type>
+  KOKKOS_INLINE_FUNCTION Type team_scan(const Type& value) const {
+    return this->template team_scan<Type>(value, nullptr);
+  }
+
+  //----------------------------------------
+
+  template <typename ReducerType>
+  KOKKOS_INLINE_FUNCTION static
+      typename std::enable_if<is_reducer<ReducerType>::value>::type
+      vector_reduce(ReducerType const& reducer) {
+    vector_reduce(reducer, reducer.reference());
+  }
+
+  template <typename ReducerType>
+  KOKKOS_INLINE_FUNCTION static
+      typename std::enable_if<is_reducer<ReducerType>::value>::type
+      vector_reduce(ReducerType const& /*reducer*/,
+                    typename ReducerType::value_type& /*value*/) {
+    // FIXME_SYCL
+    Kokkos::abort("Not implemented!");
+  }
+
+  //--------------------------------------------------------------------------
+  /**\brief  Global reduction across all blocks
+   *
+   *  Return !0 if reducer contains the final value
+   */
+  template <typename ReducerType>
+  KOKKOS_INLINE_FUNCTION static
+      typename std::enable_if<is_reducer<ReducerType>::value, int>::type
+      global_reduce(ReducerType const& /*reducer*/,
+                    int* const /*global_scratch_flags*/,
+                    void* const /*global_scratch_space*/, void* const /*shmem*/,
+                    int const /*shmem_size*/) {
+    // FIXME_SYCL
+    Kokkos::abort("Not implemented!");
+  }
+
+  //----------------------------------------
+  // Private for the driver
+
+  KOKKOS_INLINE_FUNCTION
+  SYCLTeamMember(void* shared, const int shared_begin, const int shared_size,
+                 void* scratch_level_1_ptr, const int scratch_level_1_size,
+                 const sycl::group<1> group)
+      : m_team_reduce(shared),
+        m_team_shared(((char*)shared) + shared_begin, shared_size,
```

Comment:
Have we decided on what to do about C-style casts?  (I just want to be consistent)